app/models/transformation_mapping.rb

```diff
@@ -6,4 +6,72 @@ class TransformationMapping < ApplicationRecord
   def destination(source)
     transformation_mapping_items.find_by(:source => source).try(:destination)
   end
+
+  def validate_vms(csv_list = nil)
+    csv_list.present? ? identify_vms(csv_list) : select_vms
+  end
+
+  private
+
+  def select_vms
+    # TBD
+  end
+
+  def identify_vms(csv_list)
+    valid_list = []
+    invalid_list = []
+    conflict_list = []
+
+    csv_list.each do |row|
+      vm_name = row['Name']
+
+      conditions = {:name => vm_name}
+
+      # construct conditions or selections based on MIQ exported csv columns
+      conditions[:uid_ems] = row['UID'] if row['UID'].present?
```

Comment:
This feels bad to have the API use arbitrary case for keys.  Just use lower case for everything, and, I would go farther to say that it should just use uid_ems as the key name to avoid confusion, but that could be discussed.

Reply:
All the keys are the headers from the csv, which can be an exported from manageiq, or directly created by user through an external system. All the headers exported from manageiq are capitalized. Examples are `Name`, `Provider`, `Cluster`, `IP Addresses`, `Allocated Size` etc. All columns are optional. We will do our best to identify the VMs according to provided columns.

Is `uid_ems` a good name for users who do not use manageiq exports?

My original thoughts is for the UI to read the csv file and simply convert to JSON without any modification, and pass to backend. Backend will select the columns of interest. 

Do you think the UI should select columns of interest and convert the headers to low case? In the future if there is new column added, then both UI and backend need to make changes.

Reply:
What are "MIQ exported csv columns"?  Do we generate this in reporting or something?  If it's based on our data, why wouldn't we just use our id's?

Reply:
This is an existing miq csv export functionality. When you browse an inventory of VMs you have an option to download the VMs in a CSV file. It does not contain our Id's.
But this is only one way to provide the csv file for v2v migration use. User can choose to create their own csv file using whatever method they want.